clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackServerGroupCachingAgent.groovy

```diff
@@ -280,4 +307,80 @@ class OpenstackServerGroupCachingAgent extends AbstractOpenstackCachingAgent {
 
     result
   }
+
+  @Override
+  boolean handles(OnDemandAgent.OnDemandType type, String cloudProvider) {
+    type == ServerGroup && cloudProvider == ID
+  }
+
+  @Override
+  OnDemandAgent.OnDemandResult handle(ProviderCache providerCache, Map<String, ? extends Object> data) {
+    OnDemandAgent.OnDemandResult result = null
+
+    if (data.containsKey("serverGroupName") && data.account == accountName && data.region == region) {
+      result = new OnDemandAgent.OnDemandResult(sourceAgentType: onDemandAgentType, evictions: [:].withDefault { _ -> [] })
+
+      String serverGroupName = data.serverGroupName.toString()
+      String serverGroupKey = Keys.getServerGroupKey(serverGroupName, accountName, region)
+
+      Stack stack = metricsSupport.readData {
+        try {
+          clientProvider.getStack(region, serverGroupName)
+        } catch (OpenstackProviderException ope) {
+          //Do nothing ... Exception is thrown if a stack isn't found
```

Comment:
Yep ... That is the reason for my comment.  Let me know if it could be clearer :)